src/polly/src/event_manager.rs

```diff
@@ -99,7 +100,14 @@ impl EventManager {
         }
     }
 
+    /// Register a new subscriber. All events that the subscriber is interested are registered.
+    ///
+    // TODO: Remove this workaround method. The desired state in the future is for each
+    // subscriber to call `register` directly when it needs to register an event and not have
+    // all events registered at once. This way we can also remove the `interest_list` which is
+    // only used once in this function.
     pub fn add_subscriber(&mut self, subscriber: Arc<Mutex<dyn Subscriber>>) -> Result<()> {
+        // Unwrapping here is safe because we clone a valid subscriber.
```

Comment:
It'd be more accurate to say:
```suggestion
        // Unwrapping here is safe since we want to crash on a poisoned lock.
```